zap/src/main/java/org/zaproxy/zap/view/TabbedPanelTab.java

```diff
@@ -229,6 +249,40 @@ public void setEnabled(boolean enabled) {
         }
     }
 
+    private void moveToNewWindow() {
+        int prevTabIndex = parent.indexOfComponent(component);
+        var frame = new DetachedTabFrame(title, component);
```

Comment:
Does this still work properly if other tabs are shown/added or hidden/removed (ex: installation of an add-on, changing the pin 📌 options, or actually closing ❌ a tab) while a tab is windowed.

Reply:
Yes, it does. When a tab is windowed, it's removed from the original tabbed pane, so it can only be closed by closing the window, and closing the window brings it back to the tabbed pane. Here's a quick demo :) 

https://github.com/zaproxy/zaproxy/assets/16446369/7dff3c1f-e552-4eb7-a15b-508b751d4142

Reply:
No I mean back in the mainframe the tabs list can change so the index might end-up wrong?

Reply:
Ex what if you have request, response, quick start, Neonmarker. Send Neonmarker to a window. Close quick start. Close Neonmarker window (returning it to the tab panel). Etc scenarios like that.